src/guide/receive_onchain.md

```diff
@@ -459,43 +455,39 @@ When the amount to be received exceeds the inbound liquidity of the node, a new
 To calculate the fees for a channel being opened by the LSP:
 
 ```python
-def calculate_channel_opening_fees(amount_sats):
+def calculate_channel_opening_fees(amount_msats):
     is_channel_opening_fee_needed = is_channel_opening_fee_needed()
 
     if is_channel_opening_fee_needed:
-        return calculate_fees_for_amount(amount_sats)
+        return calculate_fees_for_amount(amount_msats)
     else: 
         return None
 ```
 
 How to detect if open channel fees are needed.
 
 ```python
-def is_channel_opening_fee_needed(amount_sats):
+def is_channel_opening_fee_needed(amount_msats):
     liqudity = sdk_services.node_info().inbound_liquidity_msats // 1000
-    return amount_sats >= liqudity
+    return amount_msats >= liqudity
 ```
 
 LSP fees are calculated in two ways, either by a minimum fee set by the LSP or by a fee per myriad calculated based on the amount being received. If the fee calculated from the fee per myriad is less than the minimum fee, the minimum fee is used.
 
 This information can be retrieved for each LSP and then calculated:
 
 ```python
-def calculate_fees_for_amount(amount_sats):
+def calculate_fees_for_amount(amount_msats):
     # We need to open channel so we are calculating the fees for the LSP.
     lsp_id = sdk_services.lsp_id()
     lsp_info = sdk_services.fetch_lsp_info()
 
-    # setup_fee is the proportional fee charged based on the amount.
-    setup_fee = lsp_info.channel_fee_permyriad / 100
+    # We calculate the dynamic fees in millisatoshis rounded to satoshis.
+    channel_dynamic_fee = amount_msats * lsp_info.channel_minimum_fee_msat * lsp_info.channel_fee_permyriad / 10000 // 10000 * 10000
 
-    # min_fee is the minimum fee required by the LSP to open a channel.
-    min_fee = lsp_info.channel_minimum_fee_msat // 1000
+    fee_sat = max(lsp_info.channel_minimum_fee_msat, channel_dynamic_fee) / 1000
```

Comment:
Why is fee_sat divided by 1000 here?

Reply:
i forgot to change this, thank you. removed.